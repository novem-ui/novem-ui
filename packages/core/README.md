# @novem-ui/core

This lib exports all the Novem UI components