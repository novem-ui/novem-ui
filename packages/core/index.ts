export { default as NovemProvider } from './src/novem-provider'

export { default as Badge } from '@novem-ui/badge'
export * from '@novem-ui/badge'

export { default as Button, ButtonProps } from '@novem-ui/button'

export { default as Switch, SwitchProps } from '@novem-ui/switch'

export { Heading, Label, Paragraph, HeadingProps, LabelProps, ParagraphProps } from '@novem-ui/text'

export { Input, TextField, TextFieldProps } from '@novem-ui/input'

export { default as theme } from '@novem-ui/theme'
export type { Theme } from '@novem-ui/theme'
