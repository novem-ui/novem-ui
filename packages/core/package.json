{
  "name": "@novem-ui/core",
  "version": "0.0.1-alpha.1",
  "description": "Core components",
  "homepage": "https://github.com/jdrets/novem-ui/tree/main/packages/core#readme",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/jdrets/novem-ui.git"
  },
  "publishConfig": {
    "access": "public",
    "directory": "../../dist/core"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "yarn build:esm && yarn build:types && yarn build:package-json",
    "build:esm": "../../node_modules/.bin/babel . --root-mode upward --extensions .ts,.tsx -d ../../dist/core --source-maps",
    "build:types": "tsc -p ./tsconfig.types.json",
    "build:package-json": "../../scripts/create-package.js"
  },
  "bugs": {
    "url": "https://github.com/jdrets/novem-ui/issues"
  },
  "dependencies": {
    "@novem-ui/badge": "^0.0.1-alpha.1",
    "@novem-ui/theme": "^0.0.1-alpha.1"
  },
  "devDependencies": {
    "@types/novem-theme": "^0.0.1-alpha.1",
    "@types/react": "^17.0.3"
  },
  "peerDependencies": {
    "@emotion/react": "^11.1.5",
    "@emotion/styled": "^11.3.0",
    "polished": "^4.1.1",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
