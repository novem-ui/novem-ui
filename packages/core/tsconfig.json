{
  "extends": "../../tsconfig.json",
  "files": [],
  "include": ["**/*.ts", "**/*.tsx", "../../emotion.d.ts"],
  "exclude": ["node_modules/**", "dist/**"]
}
