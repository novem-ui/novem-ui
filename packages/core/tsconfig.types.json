{
  "extends": "../../tsconfig.json",
  "files": [],
  "compilerOptions": {
    "emitDeclarationOnly": true,
    "declaration": true,
    "declarationDir": "../../dist/",
    "declarationMap": true,
    "types": ["novem-theme"]
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules/**", "dist/**", "**/*.spec.tsx", "**/*.stories.tsx"]
}
