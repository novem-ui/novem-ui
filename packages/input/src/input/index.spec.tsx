import React from 'react'

import { render } from '@testing-library/react'
import { theme } from '@novem-ui/core'
import { matchers } from '@emotion/jest'

import Input from '.'

beforeAll(() => {
  expect.extend(matchers)
})

const setup = () => {
  const { container } = render(<Input theme={theme} />)

  const paragraph = container.querySelector('p')

  return { container, paragraph }
}

describe('input/input', () => {
  test('should render a themed input', () => {
    const { container } = setup()

    expect(container).toMatchSnapshot()
  })
})
