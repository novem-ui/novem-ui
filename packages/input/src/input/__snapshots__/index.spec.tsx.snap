// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input/input should render a themed input 1`] = `
.emotion-0 {
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 1px solid rgba(0, 0, 0, .25);
  border-radius: 1rem;
  box-sizing: border-box;
  height: 3rem;
  font-size: 16px;
  padding: 1rem;
  -webkit-transition: border ease-in 0.25s;
  transition: border ease-in 0.25s;
}

.emotion-0:focus,
.emotion-0:hover {
  outline: unset;
  border-color: #F1694B;
  box-shadow: 0 2px 4px 0 rgba(241,105,75,0.15);
}

.emotion-0:disabled {
  pointer-events: none;
  background-color: rgba(154,152,151,0.15);
}

.emotion-0:-moz-read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

.emotion-0:read-only {
  background-color: white;
  border-color: white;
  margin-left: -2px;
  padding: 0.5rem;
  pointer-events: none;
}

<div>
  <input
    class="emotion-0 emotion-1"
  />
</div>
`;
